homeassistant/components/flux_led/light.py

```diff
@@ -253,75 +253,69 @@ def effect(self):
         for effect, code in EFFECT_MAP.items():
             if current_mode == code:
                 return effect
-
         return None
 
     async def async_turn_on(self, **kwargs):
         """Turn the specified or all lights on and wait for state."""
         await self.hass.async_add_executor_job(partial(self._turn_on,
                                                        **kwargs))
-        # The bulb needs a second to tell its new values,
-        # so we wait 2 seconds before updating
-        await sleep(2)
+        # The bulb needs a bit to tell its new values,
+        # so we wait 1 second before updating
+        await sleep(1)
 
     def _turn_on(self, **kwargs):
         """Turn the specified or all lights on."""
-        self._bulb.turnOn()
+        if not self.is_on:
```

Comment:
Is there a reason for this condition? As I said, the state can be stale so it is better to not trust it.